tests/integration/components/text-field-test.js

```diff
@@ -91,3 +91,23 @@ test('"change" changes value', function(assert) {
 
   assert.equal(this.get('value'), '');
 });
+
+test('On update calls format function', function(assert) {
+  assert.expect(2);
+  this.set('value', '');
+
+  this.set('format', (value) => {
+    assert.equal(value, '2', 'calls the format function with value');
+  });
+
+  this.render(hbs`{{text-field value=value format=format}}`);
+
+  const $input = this.$('input');
+
+  run(() => {
+    $input.val('2');
+    $input.trigger('blur');
+  });
+
+  assert.equal(this.get('value'), '');
```

Comment:
You're testing whether te format function is called, but not that the value that it return is used.
Please return a new value from the format function and check the new value is the same.